hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/metadata/SparkHoodieBackedTableMetadataWriter.java

```diff
@@ -119,68 +115,20 @@ protected void initRegistry() {
 
   @Override
   protected void commit(String instantTime, Map<MetadataPartitionType, HoodieData<HoodieRecord>> partitionRecordsMap) {
-    commitInternal(instantTime, partitionRecordsMap, Option.empty());
+    commitInternal(instantTime, partitionRecordsMap, false, Option.empty());
   }
 
+  @Override
+  protected JavaRDD<HoodieRecord> convertRecordsToWriteClientInput(HoodieData<HoodieRecord> records) {
+    return HoodieJavaRDD.getJavaRDD(records);
+  }
+
+  @Override
   protected void bulkCommit(
       String instantTime, MetadataPartitionType partitionType, HoodieData<HoodieRecord> records,
       int fileGroupCount) {
     SparkHoodieMetadataBulkInsertPartitioner partitioner = new SparkHoodieMetadataBulkInsertPartitioner(fileGroupCount);
-    commitInternal(instantTime, Collections.singletonMap(partitionType, records), Option.of(partitioner));
-  }
-
-  private void commitInternal(String instantTime, Map<MetadataPartitionType, HoodieData<HoodieRecord>> partitionRecordsMap,
```

Comment:
ok i see you have moved it to super class  so that's why you need the partitioner there. I guess yu move it to reuse in Java MDT writer?

Reply:
Yes, it turns out the implementation is almost identical for all three of our clients so I moved it to a common method